native/packages/react-native-bpk-component-text-input/src/BpkTextInput.js

```diff
@@ -16,185 +16,143 @@
+
+  componentDidUpdate() {
+    Animated.parallel([
+      Animated.timing(this.animatedValues.color, {
+        toValue: this.getColorAnimatedValue(),
+        duration: animationDurationSm,
+      }),
+      Animated.timing(this.animatedValues.labelPosition, {
+        toValue: this.getLabelPositionAnimatedValue(),
+        duration: animationDurationSm,
+      }),
+    ]).start();
   }
-  if (valid !== null) {
-    // TODO Update to use Backpack-native Icon solution once implemented https://gojira.skyscanner.net/browse/BPK-841
-
-    if (valid) {
-      if (small) {
-        iconSource = tickSmIcon;
-      } else {
-        iconSource = tickLgIcon;
-      }
-    } else if (small) {
-      iconSource = exclaimationSmIcon;
-    } else {
-      iconSource = exclaimationLgIcon;
+
+  onClearText() {
+    this.inputRef.setNativeProps({ text: '' });
+    if (this.props.onChangeText) {
+      this.props.onChangeText('');
     }
+  }
 
+  getColorAnimatedValue() {
+    return this.state.isFocused ? 1 : 0;
+  }
 
-    if (valid) {
-      style.push(styles.validOuter);
-      iconStyle.push(styles.validIcon);
-    } else {
-      style.push(styles.errorOuter);
-      iconStyle.push(styles.invalidIcon);
-    }
+  getLabelPositionAnimatedValue() {
+    return (this.props.value || this.state.isFocused) ? 0 : 1;
   }
 
-  if (innerStyle) { style.push(innerStyle); }
-
-  return (
-    <View
-      style={style}
-    >
-      <TextInput
-        underlineColorAndroid="transparent"
-        placeholderTextColor={colorGray300}
-        style={textStyle}
-        editable={!disabled}
-        value={value}
-        ref={inputRef}
-        {...rest}
-      />
-      {iconSource &&
-        <Image
-          style={iconStyle}
-          source={iconSource}
-        />
-      }
-    </View>
-  );
-};
+  render() {
+    const { isFocused } = this.state;
+    const {
+      inputRef,
+      validationMessage,
+      editable,
+      label,
+      value,
+      style: userStyle,
+      valid,
+      onFocus,
+      onBlur,
+      ...rest
+    } = this.props;
+
+    const validityIcon = valid ? <ValidIcon /> : (valid === false && <InvalidIcon />);
+    const clearButton = (
+      <BpkTouchableOverlay onPress={this.onClearText} borderRadius="pill">
+        <ClearIcon />
+      </BpkTouchableOverlay>
+    );
+
+    const animatedLabelStyle = getLabelStyle(
+      this.animatedValues.color,
+      this.animatedValues.labelPosition,
+      { value, valid, editable },
+    );
+
+    const animatedInputStyle = getInputContainerStyle(this.animatedValues.color, valid);
+
+    return (
+      <View style={[styles.container, userStyle]}>
+        <Animated.Text style={animatedLabelStyle}>{label}</Animated.Text>
+        <Animated.View style={animatedInputStyle}>
+          <TextInput
+            editable={editable}
+            value={value || ''}
+            style={styles.input}
+            onFocus={() => this.setState(() => ({ isFocused: true }), onFocus)}
```

Comment:
can we have this as it's own function somewhere? it'll look much better IMHO

Reply:
Done